internal/riverinternaltest/testfactory/test_factory.go

```diff
@@ -0,0 +1,96 @@
+// Package testfactory provides low level helpers for inserting records directly
+// into the database.
+package testfactory
+
+import (
+	"context"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/riverqueue/river/internal/rivercommon"
+	"github.com/riverqueue/river/internal/util/ptrutil"
+	"github.com/riverqueue/river/riverdriver"
+	"github.com/riverqueue/river/rivertype"
+)
+
+type JobOpts struct {
+	EncodedArgs []byte
+	FinalizedAt *time.Time
+	Metadata    []byte
+	Priority    *int
+	Queue       *string
+	ScheduledAt *time.Time
+	State       *rivertype.JobState
+}
+
+func Job(ctx context.Context, t *testing.T, exec riverdriver.Executor, opts *JobOpts) *rivertype.JobRow {
```

Comment:
we should have done this a long time ago 😆

Reply:
> we should have done this a long time ago 😆

Haha, seriously +1. Just cleaned up a whole bunch more insert test helpers in `client_test.go` too.